wire-ios-data-model/Source/Model/UserClient/UserClient.swift

```diff
@@ -700,21 +741,24 @@ public extension UserClient {
 
     /// Deletes the session between the selfClient and the given userClient
     /// If there is no session it does nothing
-    func deleteSession() throws {
-        guard
-            !isSelfClient(),
-            let context = managedObjectContext,
-            let sessionID = proteusSessionID
-        else {
-            return
+    func deleteSession() async throws {
+        guard let context = managedObjectContext else { return }
+
+        var sessionID: ProteusSessionID?
+        context.performAndWait {
+            guard !isSelfClient() else {
+                return
+            }
+            sessionID = proteusSessionID
         }
+        guard let sessionID = sessionID else { return }
```

Comment:
The `performAndWait` block can return a value

```suggestion
        guard let sessionID = context.performAndWait {
            guard !isSelfClient() else {
                return
            }
            return proteusSessionID
        } else { return }
```